js/modules/timeline.src.js

```diff
@@ -248,6 +283,7 @@ seriesType('timeline', 'line', {
         if (points) {
             xMap = series.mappedXData = this.getMappedXData();
 
+            // Apply new calculated point values.
             each(points, function (point, i) {
                 point.applyOptions({
                     x: xMap[i]
```

Comment:
Consider use the series.mappedXData and remove xMap in general.

Reply:
The way of calculating the data has completely changed now.